src/metabase/query_processor/middleware/normalize_query.clj

```diff
@@ -1,47 +1,71 @@
 (ns metabase.query-processor.middleware.normalize-query
```

Comment:
We now convert to pMBQL automatically when processing a query, then we convert back to legacy a little while later.